java/org/contikios/cooja/script/ScriptLog.java

```diff
@@ -33,9 +33,11 @@
 
 /* Morty: This interface must be public, otherwise openjdk will fail */
 
+import org.contikios.cooja.Mote;
+
 public interface ScriptLog {
     void log(String log);
-    void generateMessage(long delay, String msg);
+    void generateMessage(Mote mote, long delay, String msg);
```

Comment:
I have seen some simulation scripts calling this method directly instead of using `GENERATE_MESSAGE()`. What do you think about keeping this API as is and add another (possible internal) method that takes the mote as argument?

Reply:
The point is to not dig inside the engine in different threads, but implemented your idea through a rewrite in ScriptParser instead. No visible change for the end user.